packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m

```diff
@@ -76,6 +77,26 @@ - (NSDictionary *)constantsToExport
   return @[EXMediaLibraryDidChangeEvent];
 }
 
+UM_EXPORT_METHOD_AS(requestPermissionsAsync,
```

Comment:
Is this some forward-looking solution where in the future we will deprecate asking for permissions via `expo-permissions`? Is making this change in a random module fit in a bigger scope? (i. e. wouldn't we like to copy-and-deprecate all the methods in one PR rather than in multiple random PRs)?

Reply:
Moreover, maybe the redirection could happen on JS level?

Reply:
> Is this some forward-looking solution where in the future we will deprecate asking for permissions via expo-permissions?

Yes, this is also already implemented in other modules like Location and Calendar which provide its own methods requesting for permissions. This is in the scope of #5061 but I think it doesn't bother to implement it earlier. I already discussed it with @lukmccall who is working on the new permissions. The code of these methods will probably be changed by his PR anyway, but the signature shouldn't change.

> Moreover, maybe the redirection could happen on JS level?

I don't really want to add `expo-permissions` as a dependency of `expo-media-library`, even though `expo-permissions` are installed by `react-native-unimodules`, I guess we planned to not have such deps between unimodules.